apps/system/lockscreen/js/lockscreen.js

```diff
@@ -851,15 +852,15 @@
    */
   LockScreen.prototype.checkPassCode =
   function lockscreen_checkPassCode(passcode) {
-    var request = {
-      passcode: passcode,
-      onsuccess: this.onPasscodeValidationSuccess.bind(this),
-      onerror: this.onPasscodeValidationFailed.bind(this),
-    };
-    window.dispatchEvent(new CustomEvent(
-      'lockscreen-request-passcode-validate',
-      { detail: request }
-    ));
+    PasscodeHelper.check(passcode).then((result) => {
```

Comment:
A minor issue here is if in your design, the 'PasscodeHelper' would be the only handler of the only passcode in whole Gaia? If we have different passcodes for different components, the API may need to be changed like

    `PasscodeHelper.check(passcode, 'lockscreen')`

or something like that. But of course right now it's the only passcode in Gaia, so maybe to make it as the default one is OK.